include/openthread/srp_server.h

```diff
@@ -84,11 +84,17 @@ typedef enum
  *
  * Address mode specifies how the address and port number are determined by the SRP server and how this info is
  * published in the Thread Network Data.
+ *
+ * @warning Using the `OT_SRP_SERVER_ADDRESS_MODE_UNICAST_FORCE_ADD` option will make the implementation
+ * non-compliant with the Thread specification. This option is intended for testing and specific demo purposes.
+ * When selected, the SRP server, upon being enabled, will bypass the Network Data publisher and always add the
+ * "SRP/DNS unicast" entry directly to the Network Data, regardless of how many other similar entries are present.
  */
 typedef enum otSrpServerAddressMode
 {
-    OT_SRP_SERVER_ADDRESS_MODE_UNICAST = 0, ///< Unicast address mode.
-    OT_SRP_SERVER_ADDRESS_MODE_ANYCAST = 1, ///< Anycast address mode.
+    OT_SRP_SERVER_ADDRESS_MODE_UNICAST           = 0, ///< Unicast address mode. Use Network Data publisher.
+    OT_SRP_SERVER_ADDRESS_MODE_ANYCAST           = 1, ///< Anycast address mode. Use Network Data publisher
+    OT_SRP_SERVER_ADDRESS_MODE_UNICAST_FORCE_ADD = 2, ///< Unicast address mode. Immediately force add to Network Data.
```

Comment:
@abtink This is going to always publish the service even if there is already a SRP server? This is not we want for Thread in mobile phones - if there is already a SRP server (either a BR or another phone), this phone should not register another SRP server again (@bukepo thanks for catching this

Reply:
So I think we still need to use the netdata publisher to support "exclusive" DNS/SRP server service entry. Currently, it allows 2 unicast SRP server at the same time, can we add support for "only one SRP server at the same time"? Note that for Android, we need to support both BR use case and phone use case with the same OT binary, so we should not change `OPENTHREAD_CONFIG_NETDATA_PUBLISHER_DESIRED_NUM_UNICAST_DNS_SRP_SERVICE_ENTRIES` directly.

Reply:
My understanding was that you wanted this for the very specific situation of having a device (e.g., a mobile phone) act as a temporary BR because there are no other BRs in the network (and only when, in such a case, it is desirable for it to start its SRP server quickly using this new option).

Any device (mobile or otherwise) wanting to act as a conventional Thread BR should follow the specification and should not use this.